app/server/utils/ussd/kenya_ussd_state_machine.py

```diff
@@ -171,14 +172,30 @@ def save_transaction_amount(self, user_input):
         self.session.set_data('transaction_amount', user_input)
 
     def save_transaction_reason(self, user_input):
-        #TODO: use ruben's dynamic code to convert number to reason
-        self.session.set_data('transaction_reason_translated', user_input)
-        self.session.set_data('transaction_reason_id', "1")
+        chosen_transfer_usage = self.get_select_transfer_usage(user_input)
+        self.session.set_data('transaction_reason', chosen_transfer_usage.name)
```

Comment:
we don't use transaction_reason in the data anymore, just translated and id. we can rename translated to name if it's confusing, or transaction_reason_i18n? no strong opinion here